fork_choice/safe-block.md

```diff
@@ -0,0 +1,31 @@
+# Fork Choice -- Safe Block
+
+## Table of contents
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [`get_safe_beacon_block`](#get_safe_beacon_block)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+Under honest majority and certain network synchronicity assumptions
+there exist a block that is safe from re-orgs. Normally this block is
+pretty close to the head of canonical chain which makes it valuable
+to expose a safe block to users.
+
+This section describes an algorithm to find a safe block.
+
+## `get_safe_beacon_block`
+
+```python
+def get_safe_beacon_block(store: Store) -> Root:
+    # Use most recent justified block as a stopgap
+    return store.justified_checkpoint.root
```

Comment:
this should either be named `get_safe_beacon_block_root` and return a `Root` (as is)
OR
should be named `get_safe_beacon_block` and return the full `BeaconBlock`.

I'm okay either way

Reply:
The idea was to follow `get_head(store: Store) -> Root:` naming notation, i would expect `get_head` returning a beacon block, thought it could be argued.